### PR TITLE
Add a 'force restart' feature, including auto-generated code for uninitialized storage.

### PR DESCRIPTION
**Pull request by alexandraavendano**
**Changes:**

This PR introduces a new 'force restart' feature for bare metal instance servers. The 'force restart' feature allows users to perform an immediate reboot of a server, bypassing any graceful shutdown processes. This is particularly useful in scenarios where standard reboot procedures fail or take too long. Leveraging current standards, this capability ensures rapid recovery of a system, reducing downtime and ensuring system availability. 

It's important to note that force restart should be used judiciously, as it may result in unsaved data loss or potential file system corruption.

**Note for reviewers:** Most of the changes are autogenerated using autorest. However, the following list are the list of files that needs to be updated manually. 

- README.md with documentation of the new command(s)
- setup.py with new version number
- HISTORY.rst with the new version number, and description of the version
- azext_baremetalinstance/azext_metadata.json with new version number
- azext_baremetalinstance/_help.py with info about the new command(s)
- azext_baremetalinstance/custom.py with code calling your new command(s)
- This file includes methods that are going to be invoked by your commands. The methods are implemented using the client created using the client factory you passed in the command.py file.
- azext_baremetalinstance/commands.py with routing for the new command(s): This file creates the commands for your module. Eg: baremetalinstanceshow, baremetalinstancelist, etc. It also states which method to call when your command is invoked, as well as which client factory to use.
- azext_baremetalinstance/_params.py with new parameters for the new command(s): This file defines the parameters the commands are going to use.

Test plan:

Input: Run `az baremetalinstance list` in central using a susbcription with multiple blades assigned.
Output:
```
[
  {
    "azureBareMetalInstanceId": "5d177a4e-c221-4666-a5f4-171e815c9e98",
    "hardwareProfile": {
      "azureBareMetalInstanceSize": "S448SE",
      "hardwareType": "SDFLEX"
    },
    "hwRevision": "Rev 4.2",
    "id": "/subscriptions/10267def-42d5-4bc8-8cc0-887da4742d06/resourceGroups/DSM05A-T210/providers/Microsoft.BareMetalInfrastructure/bareMetalInstances/epiciotooltest1",
    "location": "centraluseuap",
    "name": "epiciotooltest1",
    "networkProfile": {
      "circuitId": "/subscriptions/cf471ca4-2535-48fb-b676-684d8f7bfc6e/resourceGroups/T210_DSM05-Circuits/providers/Microsoft.Network/expressRouteCircuits/T210_DSM05-10Gb",
      "networkInterfaces": [
        {
          "ipAddress": "10.100.0.91"
        }
      ]
    },
    "osProfile": {
      "computerName": "epiciotooltest1",
      "osType": "RHEL 8.4",
      "sshPublicKey": null,
      "version": "8.4"
    },
    "partnerNodeId": null,
    "powerState": "started",
    "provisioningState": "Succeeded",
    "proximityPlacementGroup": "/subscriptions/10267def-42d5-4bc8-8cc0-887da4742d06/resourcegroups/DSM05A-T210/providers/Microsoft.Compute/proximityPlacementGroups/epiciotooltest1-ppg",
    "resourceGroup": "DSM05A-T210",
    "storageProfile": {
      "nfsIpAddress": "10.20.211.91",
      "osDisks": [
        {
          "diskSizeGb": 50,
          "lun": null,
          "name": "t210_rhel_boot_epiciotooltest1_vol_lun"
        }
      ]
    },
    "systemData": {
      "lastModifiedAt": "2023-10-23T18:19:20.7198608Z",
      "lastModifiedBy": "aavendano@microsoft.com",
      "lastModifiedByType": "User"
    },
    "tags": {},
    "type": "Microsoft.BareMetalInfrastructure/bareMetalInstances"
  }
]
```
Input:  Run `az baremetalinstance restart --resource-group DSM05A-T210 --instance-name epiciotooltest1` 
Output:
```
{
  "endTime": "2023-10-31T18:03:58Z",
  "name": "818e9bd1-f474-45d6-a56a-4308e7e07d09",
  "startTime": "2023-10-31T18:00:54.6822673Z",
  "status": "Succeeded"
}
```

### CODE REVIEW

File: azext_baremetalinfrastructure/_params.py [hunk 1]
```diff
@@ -3,12 +3,19 @@
...
-
-from azure.cli.core.commands.parameters import resource_group_name_type
+from azure.cli.core.commands.parameters import (
+    resource_group_name_type,
+    get_three_state_flag
+)
 
 
 def load_arguments(self, _):
+
```
Comment (pabowers) [inline azext_baremetalinfrastructure/_params.py:13]: why do we want an empty line after the function definition?

**Comment by pabowers** [comment]
The PR description mentions azext_hanaonazure instead of azext_baremetalinfrastructure. Probably a typo, but please update accordingly

**Comment by pabowers** [comment]
I think this `azext_hanaonazure/_help.py` is missing from the PR

File: azext_baremetalinfrastructure/custom.py [hunk 2]
```diff
@@ -15,14 +15,16 @@ def list_baremetalinstance(client, resource_group_name=None):
         return client.list_by_subscription()
     return client.list(resource_group_name)
 
-def restart_baremetalinstance(client, resource_group_name, instance_name):
-    # The restart baremetalinstance REST API is a POST with no body.
+
```
Comment (pabowers) [inline azext_baremetalinfrastructure/custom.py:18]: nit: we don't need the extra empty lines before or after the function

File: azext_baremetalinfrastructure/custom.py [hunk 3]
```diff
@@ -33,6 +35,7 @@ def start_baremetalinstance(client, resource_group_name, instance_name):
     custom_header['Content-Type'] = 'application/json; charset=utf-8'
     return client.begin_start(resource_group_name, instance_name, headers=custom_header)
 
+
```
Comment (pabowers) [inline azext_baremetalinfrastructure/custom.py:38]: why are we adding a random empty line?

**Review by pabowers — CHANGES_REQUESTED** [review]
Overall on the right track, just a few areas for feedback.

**Comment by alexandraavendano** [comment]
> I think this `azext_baremetalinfrastructure/_help.py` is missing from the PR

In this case I used `help="Force restart option"` in _params.py L21. This is allowing me to show the following information when we use -h:
image

**Review by pabowers — APPROVED** [review]
LGTM

**Review by 8Gitbrix — APPROVED** [review]

File: azext_baremetalinfrastructure/custom.py [hunk 4]
```diff
@@ -5,24 +5,22 @@
...
 
-def restart_baremetalinstance(client, resource_group_name, instance_name):
-    # The restart baremetalinstance REST API is a POST with no body.
+def restart_baremetalinstance(client, resource_group_name, instance_name, force=False):
+    # The restart baremetalinstance REST API is a POST with a body.
     # The HaaS RP API requires the Content-Type to be set.
     # Swagger does not add the Content-Type in the header if there is no body.
     # We need to add a custom header to force the API call to add the Content-Type.
     custom_header = {}
-    custom_header['Content-Type'] = 'application/json; charset=utf-8'
```
Comment (joacocruz6) [inline azext_baremetalinfrastructure/custom.py:24]: are this not necessary anymore?

Comment (alexandraavendano) [reply azext_baremetalinfrastructure/custom.py:24]: The custom header is used in the `begin_restart` function, who is calling `_restart_initial`. Inside restart initial we have the following code:

image

meaning that for empty object we by default are going to be granted the original "application/json".

**Review by joacocruz6 — APPROVED** [review]

**Pull request merged**